core/server/master/src/main/java/alluxio/master/metastore/rocks/RocksBlockMetaStore.java

```diff
@@ -318,8 +338,17 @@ public void removeBlock(long id) {
 
   @Override
   public void clear() {
-    mSize.reset();
-    mRocksStore.clear();
+    // Block all new readers and make concurrent readers bail asap
+    LOG.info("Marking RocksDB closed so all concurrent read/write should stop");
+    mClosed.set(true);
+    try (LockResource lock = new LockResource(mDbStateLock.writeLock())) {
+      LOG.info("Clearing RocksDB");
+      mSize.reset();
+      mRocksStore.clear();
+    }
+    // Reset the DB state and prepare to serve again
+    LOG.info("RocksDB ready to serve again");
+    mClosed.set(false);
```

Comment:
TODO: even though the DB is up again, previous iterators do not make sense anymore...